<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PCLViewer</class>
 <widget class="QMainWindow" name="PCLViewer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1980</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>5000</width>
    <height>5000</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>PCLViewer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QVTKWidget" name="qvtkWidget" native="true">
    <property name="geometry">
     <rect>
      <x>670</x>
      <y>20</y>
      <width>1100</width>
      <height>640</height>
     </rect>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalSlider_x1">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>170</y>
      <width>160</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-200</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="singleStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>-50</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalSlider_y1">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>250</y>
      <width>160</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-200</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>-40</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalSlider_z1">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>330</y>
      <width>160</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-200</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>130</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>x bottom</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>210</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>y bottom</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>300</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>z bottom</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_save">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>440</y>
      <width>181</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>save</string>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalSlider_x2">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>170</y>
      <width>160</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-200</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>50</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>300</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>z upper</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>210</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>y upper</string>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalSlider_y2">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>250</y>
      <width>160</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-200</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>25</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>130</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>x upper</string>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalSlider_z2">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>330</y>
      <width>160</width>
      <height>29</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-200</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>150</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>50</y>
      <width>271</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>pointcloud conditional filter</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>380</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>voxel-leafsize</string>
    </property>
   </widget>
   <widget class="QSlider" name="horizontalSlider_z1_2">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>410</y>
      <width>160</width>
      <height>29</height>
     </rect>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="value">
     <number>128</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox_x1">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>170</y>
      <width>48</width>
      <height>27</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-200</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="singleStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>-50</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox_y1">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>250</y>
      <width>48</width>
      <height>27</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-200</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>-40</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox_z1">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>330</y>
      <width>48</width>
      <height>27</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-200</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox_z2">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>330</y>
      <width>48</width>
      <height>27</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-200</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>150</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox_y2">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>250</y>
      <width>48</width>
      <height>27</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-200</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>25</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox_x2">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>170</y>
      <width>48</width>
      <height>27</height>
     </rect>
    </property>
    <property name="minimum">
     <number>-200</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="value">
     <number>50</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_1">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>170</y>
      <width>41</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>-0.5</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>250</y>
      <width>41</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>-0.4</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>180</y>
      <width>41</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>0.5</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>340</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>340</y>
      <width>41</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>1.5</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>260</y>
      <width>41</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>0.25</string>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QVTKWidget</class>
   <extends>QWidget</extends>
   <header>QVTKWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>horizontalSlider_x1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox_x1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>82</x>
     <y>175</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>182</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_x1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider_x1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>206</x>
     <y>187</y>
    </hint>
    <hint type="destinationlabel">
     <x>66</x>
     <y>186</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_y1</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>spinBox_y1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>100</x>
     <y>262</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>250</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_y1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider_y1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>227</x>
     <y>269</y>
    </hint>
    <hint type="destinationlabel">
     <x>144</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_z1</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>spinBox_z1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_z1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider_z1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>215</x>
     <y>353</y>
    </hint>
    <hint type="destinationlabel">
     <x>174</x>
     <y>349</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_x2</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>spinBox_x2</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>430</x>
     <y>188</y>
    </hint>
    <hint type="destinationlabel">
     <x>505</x>
     <y>189</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_y2</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>spinBox_y2</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>437</x>
     <y>261</y>
    </hint>
    <hint type="destinationlabel">
     <x>505</x>
     <y>261</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_z2</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>spinBox_z2</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>451</x>
     <y>342</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>345</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_x2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider_x2</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>507</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>461</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_y2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider_y2</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>510</x>
     <y>251</y>
    </hint>
    <hint type="destinationlabel">
     <x>457</x>
     <y>256</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_z2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider_z2</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>514</x>
     <y>344</y>
    </hint>
    <hint type="destinationlabel">
     <x>460</x>
     <y>340</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
